scaling/src/test/java/io/quarkus/ts/openshift/scaling/ScalingOpenShiftIT.java

```diff
@@ -38,91 +43,81 @@ public void scaleBack() {
-                    given()
-                            .when().get("/scaling")
-                            .then()
-                            .statusCode(OK.getStatusCode())
-                            .extract().asString();
-            uniqueIds.add(uniqueId);
-
-            assertThat(uniqueIds).hasSize(2);
-        });
-
-        openShiftUtil.scale(DC_NAME, 1);
-
-        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(1);
-
-        uniqueIds.clear();
-        for (int i = 0; i < 100; i++) {
-            String uniqueId =
-                    given()
-                            .when().get("/scaling")
-                            .then()
-                            .statusCode(OK.getStatusCode())
-                            .extract().asString();
-            uniqueIds.add(uniqueId);
-        }
-
-        assertThat(uniqueIds).hasSize(1);
-
-        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(1);
+    public void scaleDownTest() throws InterruptedException {
+
+        int REPLICAS = EXPECTED_SCALE_UP;
+        givenResourcePath("/scaling");
+        whenScaleTo(EXPECTED_SCALE_UP);
+        thenCheckReplicasAmount(EXPECTED_SCALE_UP);
+        whenMakeRequestTo("/scaling", REPLICAS);
+
+        REPLICAS = EXPECTED_SCALE_DOWN;
+        whenScaleTo(EXPECTED_SCALE_DOWN);
+        thenCheckReplicasAmount(EXPECTED_SCALE_DOWN);
+        whenMakeRequestTo("/scaling", REPLICAS);
     }
 
+
     /**
      * Workflow:
      * * Scale down to zero replicas.
      * * Execute an arbitrary minimal sample of requests and verify that all get HTTP 503 response.
      */
     @Test
-    public void scaleToZero() {
-        openShiftUtil.scale(DC_NAME, 0);
+    public void scaleToZero() throws InterruptedException {
+
+        givenResourcePath("/scaling");
+        whenScaleTo(0);
+        thenCheckReplicasAmount(0);
+        makeHttpScalingRequest("/scaling", SERVICE_UNAVAILABLE.getStatusCode());
+    }
+
+    private void givenResourcePath(String path) {
+        given()
+                .when().get("/scaling")
+                .then()
+                .statusCode(OK.getStatusCode());
+    }
+
+    private void whenScaleTo(int amount) {
+        openShiftUtil.scale(DC_NAME, amount);
+    }
 
-        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(0);
+    private void thenCheckReplicasAmount(int expectedAmount) {
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(expectedAmount);
+    }
+
+    private void whenMakeRequestTo(String path, int expectedReplicas) {
+        Set<String> replicas = new HashSet<>();
+        await().pollInterval(DELAY_BETWEEN_REQUEST_MS, TimeUnit.MILLISECONDS)
+                .atMost(TIMEOUT_SEC, TimeUnit.SECONDS)
+                .untilAsserted(() -> {
+                    replicas.add(makeHttpScalingRequest(path, OK.getStatusCode()).extract().asString());
+                    assertEquals(replicas.size(), expectedReplicas);
```

Comment:
The expected value should be the first argument, otherwise the message in case of assertion error is confusing.